packages/components/Menu/src/MenuItemRadio/useMenuItemRadio.ts

```diff
@@ -13,7 +13,7 @@ export const useMenuItemRadio = (props: MenuItemCheckboxProps): MenuItemCheckbox
   const listContext = useMenuListContext();
   const selectRadio = listContext.selectRadio;
   const setOpen = context.setOpen;
-  let shouldPersist = context.persistOnItemClick;
+  let shouldPersist = context.persistOnItemClick || context.isControlled; // Don't close the Menu if it is controlled Menu if MenuItemRadio is clicked
```

Comment:
The way the code should be working is that setOpenInternal shouldn't be called in useMenu if the menu is controlled. This would prevent the menu from closing. If we block the event here then the parent component won't get notified that the component got an action that should close it.